Katello/katello
# Fixes #30641 - CV publish fails if checksum type changes

Fixes untested erroneous code that was running if a user changes their repo's checksum type before publishing it in a content view.

To test:
1) Make sure you're using Pulp 3 for yum content
2) Create and sync a repo
3) Add that repo to a content view
4) Publish the content view
5) Change the repo's checksum type to sha1
6) Publish the content view again.  This will fail without my PR.
